File: ContactList/app/src/main/java/com/bergamin/contactlist/FormHelper.java
package com.bergamin.contactlist;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;

import com.bergamin.contactlist.model.Contact;

/**
 * Created by Guilherme on 02/04/2016.
 */
public class FormHelper {

    private final TextView nameTxt;
    private final TextView addressTxt;
    private final TextView phoneTxt;
    private final TextView websiteTxt;
    private final ImageView photoImg;
    private Contact contact;

    public FormHelper(FormActivity activity){
        this.contact = new Contact();
        nameTxt = (TextView) activity.findViewById(R.id.nameTxt);
        addressTxt = (TextView) activity.findViewById(R.id.addressTxt);
        phoneTxt = (TextView) activity.findViewById(R.id.phoneTxt);
        websiteTxt = (TextView) activity.findViewById(R.id.websiteTxt);
        photoImg = (ImageView) activity.findViewById(R.id.photoImg);
    }

    public Contact getContact(){

        contact.setName(nameTxt.getText().toString());
        contact.setAddress(addressTxt.getText().toString());
        contact.setPhone(phoneTxt.getText().toString());
        contact.setWebSite(websiteTxt.getText().toString());
        contact.setPhotoPath((String) photoImg.getTag());

        return contact;
    }

    public void fillForm(Contact contact) {
        nameTxt.setText(contact.getName());
        addressTxt.setText(contact.getAddress());
        phoneTxt.setText(contact.getPhone());
        websiteTxt.setText(contact.getWebSite());
        loadImage(contact.getPhotoPath());
        this.contact = contact;
    }

    public void loadImage(String filePath) {
        if(filePath != null) {
            Bitmap bitmap = BitmapFactory.decodeFile(filePath);
            bitmap = Bitmap.createScaledBitmap(bitmap, 300, 300, true);
            photoImg.setImageBitmap(bitmap);
            photoImg.setScaleType(ImageView.ScaleType.FIT_XY);
            photoImg.setTag(filePath);
        }
    }
}

File: Kotlin/ContactList/app/src/main/java/com/bergamin/contactlist/FormHelper.kt
package com.bergamin.contactlist

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.widget.ImageView
import android.widget.TextView
import com.bergamin.contactlist.model.Contact

/**
 * Created by gbergamin on 24/11/2017.
 */
class FormHelper(
        var activity: FormActivity,
        var contact: Contact) {

    val nameTxt: TextView
    val addressTxt: TextView
    val phoneTxt: TextView
    val websiteTxt: TextView
    val photoImg: ImageView

    init {
        nameTxt = activity.findViewById(R.id.nameTxt)
        addressTxt = activity.findViewById(R.id.addressTxt)
        phoneTxt = activity.findViewById(R.id.phoneTxt)
        websiteTxt = activity.findViewById(R.id.websiteTxt)
        photoImg = activity.findViewById(R.id.photoImg)

        contact.name = nameTxt.text.toString()
        contact.address = addressTxt.text.toString()
        contact.phone = phoneTxt.text.toString()
        contact.webSite = websiteTxt.text.toString()
        contact.photoPath = photoImg.tag as String
    }
    fun fillForm(contact: Contact){
        nameTxt.text = contact.name
        addressTxt.text = contact.address
        phoneTxt.text = contact.phone
        websiteTxt.text = contact.webSite

        loadImage(contact.photoPath)
        this.contact = contact
    }

    fun loadImage(path: String) {
        var bitmap = BitmapFactory.decodeFile(path)
        bitmap = Bitmap.createScaledBitmap(bitmap,300,300, true)

        photoImg.setImageBitmap(bitmap)
        photoImg.scaleType = ImageView.ScaleType.FIT_XY
        photoImg.tag = path
    }
}